### PR TITLE
New package: RandyHollines.Objeck version 23.11.2

### LINKED ISSUE #112281: [Package Request]: Objeck (language)
### How can we help?

I would like someone else to build the manifest.

### Please read and ensure the following

- [X] The installer meets the above requirements

### Please provide the following information

Download Page Url: https://objeck.org/
Publisher: 
Package Name: objekt (or object-lang)
Description: General-purpose object-oriented and functional programming language. Designed to be intuitive, lightweight, cross-platform and fast.
Package Version:  2023.7.2
Installer URL:  https://github.com/objeck/objeck-lang/releases/download/v2023.7.2/objeck-windows-x64_2023.7.2.msi and https://github.com/objeck/objeck-lang/releases/download/v2023.7.2/objeck-windows-x64_2023.7.2.zip

### PR DESCRIPTION
**Pull request by Dragon1573**
## Description

- Resolve #112281

## Checks

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.5 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.5.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/129725)

### CODE REVIEW

**Comment by Dragon1573** [comment]
@stephengillie - I'm testing the package using Windows Sandbox. On my device, it's able to launch the app using shortcut in Start Menu. A "central panel" will launch, provides such as Objeck Command Prompt, API Documentatuon and Examples.

I'm unable to reproduce the issue you found. :confused:

**Comment by Dragon1573** [comment]
<details><summary>Local Powershell</summary>

```text
D:\Workloads\winget-pkgs [issues/112281 ≡]> .\Tools\SandboxTest.ps1 .\manifests\r\RandyHollines\Objeck\23.11.2\
--> Validating Manifest
清单验证成功。

--> Checking dependencies
--> Starting Windows Sandbox, and:
    - Mounting the following directories:
      - C:\Users\Administrator\AppData\Local\Temp\SandboxTest as read-only
      - D:\Workloads\winget-pkgs as read-and-write
    - Installing WinGet
    - Configuring Winget
    - Installing the Manifest 23.11.2
    - Refreshing environment variables
    - Comparing ARP Entries
```

</details>

<details><summary>Powershell in Windows Terminal</summary>

```text
--> Installing WinGet
--> Disabling safety warning when running installer
Tip: you can type 'Update-EnvironmentVariables' to update your environment variables, such as after installing a new software.

--> Configuring Winget
Enabled admin setting 'LocalManifestFiles'.
Enabled admin setting 'LocalArchiveMalwareScanOverride'.


--> Installing the Manifest 23.11.2

Found Objeck [RandyHollines.Objeck] Version 23.11.2
This application is licensed to you by its owner.
Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
Downloading https://github.com/objeck/objeck-lang/releases/download/v2023.11.2/objeck-windows-x64_v2023.11.2.msi
  ██████████████████████████████  15.7 MB / 15.7 MB
Successfully verified installer hash
Starting package install...
Successfully installed

--> Refreshing environment variables

--> Comparing ARP Entries

DisplayName DisplayVersion Publisher      ProductCode                            Scope
----------- -------------- ---------      -----------                            -----
Objeck      23.11.2        Randy Hollines {C6FF1E2B-F4A3-420F-89D0-FBFFE7069646} Machine
```

</details>

<details><summary>Launch with icon in Start Menu</summary>

image

image

</details>

**Comment by Dragon1573** [comment]
@stephengillie - Still unable to reproduce the issue ... 😕

PixPin_2023-12-13_15-09-57

**Comment by stephengillie** [comment]
Found the issue - a missing dependency:
image

File: manifests/r/RandyHollines/Objeck/23.11.2/RandyHollines.Objeck.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Created using wingetcreate 1.5.7.0
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.5.0.schema.json
+
+PackageIdentifier: RandyHollines.Objeck
+PackageVersion: 23.11.2
+InstallerLocale: en-US
+InstallerType: msi
+ProductCode: '{C6FF1E2B-F4A3-420F-89D0-FBFFE7069646}'
```
Comment (stephengillie) [inline manifests/r/RandyHollines/Objeck/23.11.2/RandyHollines.Objeck.installer.yaml:8]:
```suggestion
ProductCode: '{C6FF1E2B-F4A3-420F-89D0-FBFFE7069646}'
Dependencies:
  PackageDependencies:
    - PackageIdentifier: Microsoft.VCRedist.2015+.x64
```

**Review by stephengillie — APPROVED** [review]

**Pull request merged**